include/ht_material_resource.h

```diff
@@ -9,16 +10,15 @@ namespace Hatchit
         class HT_API Material : public Resource<Material>
         {
         public:
-            Material(std::string name);
+            Material(std::string filename);
             virtual ~Material() = default;
 
-
-            virtual bool VInitFromFile(const std::string& file) override;
+            virtual bool VInitFromFile(const std::string& filename) override;
 
         private:
             std::string m_pipeline;
             std::string m_renderPass;
-            std::vector<std::string> m_shaderVariables;
+            std::map<std::string, ShaderVariable*> m_shaderVariables;
```

Comment:
Looking at the current file specification, we can either do a map with names, or a vector of ShaderVariables.  I guess it depends more on how we're accessing variables (name vs. index).